File: Editorial/84_Largest_Rectangle_in_Histogram/Go/2_Better_Brute_Force.go
func largestRectangleArea(heights []int) int {
    maxArea := 0
    length := len(heights)
    for i := 0; i < length; i++ {
        minHeight := int(^uint(0) >> 1) // set to maximum int value
        for j := i; j < length; j++ {
            if minHeight > heights[j] {
                minHeight = heights[j]
            }
            area := minHeight * (j - i + 1)
            if maxArea < area {
                maxArea = area
            }
        }
    }
    return maxArea
}

File: Editorial/84_Largest_Rectangle_in_Histogram/CSharp/2_Better_Brute_Force.cs
public class Solution {
    public int LargestRectangleArea(int[] heights) {
        int maxArea = 0;
        int length = heights.Length;
        for (int i = 0; i < length; i++) {
            int minHeight = int.MaxValue;
            for (int j = i; j < length; j++) {
                minHeight = Math.Min(minHeight, heights[j]);
                maxArea = Math.Max(maxArea, minHeight * (j - i + 1));
            }
        }

        return maxArea;
    }
}
